api/src/test/java/bio/terra/tanagra/app/EntitiesFilterApiControllerTest.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.tanagra.app;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.tanagra.app.controller.EntitiesFiltersApiController;
+import bio.terra.tanagra.generated.model.ApiAttributeValue;
+import bio.terra.tanagra.generated.model.ApiAttributeVariable;
+import bio.terra.tanagra.generated.model.ApiBinaryFilter;
+import bio.terra.tanagra.generated.model.ApiBinaryFilterOperator;
+import bio.terra.tanagra.generated.model.ApiEntityFilter;
+import bio.terra.tanagra.generated.model.ApiFilter;
+import bio.terra.tanagra.generated.model.ApiSqlQuery;
+import bio.terra.tanagra.service.underlay.NauticalUnderlayUtils;
+import bio.terra.tanagra.testing.BaseSpringUnitTest;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("nautical")
+public class EntitiesFilterApiControllerTest extends BaseSpringUnitTest {
+  @Autowired private EntitiesFiltersApiController apiController;
+
+  @Test
+  void generateSqlQuery() {
+    ApiEntityFilter apiEntityFilter =
+        new ApiEntityFilter()
+            .entityVariable("s")
+            .filter(
+                new ApiFilter()
+                    .binaryFilter(
+                        new ApiBinaryFilter()
+                            .attributeVariable(
+                                new ApiAttributeVariable().variable("s").name("rating"))
+                            .operator(ApiBinaryFilterOperator.EQUALS)
+                            .attributeValue(new ApiAttributeValue().int64Val(42L))));
+
+    ResponseEntity<ApiSqlQuery> response =
```

Comment:
+1 to not mocking the services here. I'm a fan of this kind of controller test, i.e. just using the real underlying services; if possible down to in-memory versions of external dependencies.

Reply:
I really dislike mocking :) It's at best an occasionally necessary evil. But after a doing a few refactors on code with mocks for "test coverage," their value seems extremely limited.